packages/next-swc/crates/next-core/src/next_shared/transforms/swc_ecma_transform_plugins.rs

```diff
@@ -0,0 +1,104 @@
+use anyhow::Result;
+use turbo_tasks_fs::FileSystemPathVc;
+use turbopack_binding::turbopack::ecmascript::OptionTransformPluginVc;
+
+use crate::next_config::NextConfigVc;
+
+#[turbo_tasks::function]
+pub async fn get_swc_ecma_transform_plugin(
+    #[cfg_attr(not(feature = "plugin"), allow(unused))] project_path: FileSystemPathVc,
+    next_config: NextConfigVc,
+) -> Result<OptionTransformPluginVc> {
+    let config = next_config.await?;
+    match config.experimental.swc_plugins.as_ref() {
+        Some(plugin_configs) if !plugin_configs.is_empty() => {
+            #[cfg(feature = "plugin")]
```

Comment:
Nit: Can this be hoisted higher, so that we don't need an extra branch path?

```rust
#[cfg(not(feature = "plugin"))]
Ok(OptionTransformPluginVc::cell(None))

#[cfg(feature = "plugin")]
{ … }

Reply:
Probably yes, have to double check.